app/controllers/miq_ae_class_controller.rb

```diff
@@ -1853,13 +1854,15 @@ def delete_domain
         next unless domain
         if domain.editable_properties?
           aedomains.push(domain.id)
+          repositories.push(domain.git_repository.id) if domain.git_enabled?
         else
           add_flash(_("Read Only %{model} \"%{name}\" cannot be deleted") %
             {:model => ui_lookup(:model => "MiqAeDomain"), :name => domain.name}, :error)
         end
       end
     end
     process_elements(aedomains, MiqAeDomain, 'destroy') unless aedomains.empty?
+    repositories.each { |id| git_based_domain_import_service.destroy_repository(id) }
```

Comment:
Can this repo `destroy` happen as part of the `destroy` for the MiqAeDomain in a `before_destroy` callback?  Seems odd to make it a separate task here.  I would think you would want them linked so if the domain is called from other places, for example the API, the repo would be cleaned up as well.